### PR TITLE
feat(llmobs): add token metrics for langchain llm/chat model spans when appropriate, color as appropriate

### PR DESCRIPTION
**Pull request by sabrenner**
## Checklist
- [ ] PR author has checked that all the criteria below are met
- The PR description includes an overview of the change
- The PR description articulates the motivation for the change
- The change includes tests OR the PR description describes a testing strategy
- The PR description notes risks associated with the change, if any
- Newly-added code is easy to change
- The change follows the [library release note guidelines](https://ddtrace.readthedocs.io/en/stable/releasenotes.html)
- The change includes or references documentation updates if necessary
- Backport labels are set (if [applicable](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting))

## Reviewer Checklist
- [ ] Reviewer has checked that all the criteria below are met 
- Title is accurate
- All changes are related to the pull request's stated goal
- Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes
- Testing strategy adequately addresses listed risks
- Newly-added code is easy to change
- Release note makes sense to a user of the library
- If necessary, author has acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment
- Backport labels are set in a manner that is consistent with the [release branch maintenance policy](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting)

### CODE REVIEW

File: releasenotes/notes/llmobs-langchain-token-metrics-and-coloring-21eed854130b235c.yaml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+features:
+  - |
+    LLM Observability: When not using a provider integration (OpenAI, Anthropic, or Bedrock) with the LangChain integration, token metrics will be appended to the LLM Observability ``llm`` span.
+    LLM Observability: When ``chat_model.with_structured_output(..., method="json_mode")`` is used, or ``response_format={"type": "json_object"}`` is passed into the chat model invocation, the LLM Observability will be an ``llm`` span instead of a ``workflow`` span.
```
Comment (sabrenner) [inline releasenotes/notes/llmobs-langchain-token-metrics-and-coloring-21eed854130b235c.yaml:5]:
```suggestion
    LLM Observability: When ``chat_model.with_structured_output(..., method="json_mode")`` is used, or ``response_format={"type": "json_object"}`` is passed into the chat model invocation, the LLM Observability span will be an ``llm`` span instead of a ``workflow`` span.                 
```

File: ddtrace/llmobs/_integrations/langchain.py [hunk 2]
```diff
@@ -97,7 +103,8 @@ def _llmobs_set_tags(
         elif operation == "tool":
             self._llmobs_set_meta_tags_from_tool(span, tool_inputs=kwargs, tool_output=response)
 
-        span.set_tag_str(METRICS, safe_json({}))
+        if span.get_tag(METRICS) is None:
+            span.set_tag_str(METRICS, safe_json({}))
```
Comment (Yun-Kim) [inline ddtrace/llmobs/_integrations/langchain.py:107]: Do we need to set metrics if empty? If not, I think we can just get rid of this.

File: ddtrace/llmobs/_integrations/langchain.py [hunk 3]
```diff
@@ -441,6 +490,42 @@ def record_usage(self, span: Span, usage: Dict[str, Any]) -> None:
...
             self.metric(span, "incr", "tokens.total_cost", total_cost)
 
+    def check_token_usage_chat_or_llm_result(self, result):
+        """Checks for token usage on the top-level ChatResult or LLMResult object"""
+        llm_output = getattr(result, "llm_output", None) or {}  # in case it is explicitly set to None
+        token_usage = (
+            llm_output.get("token_usage", {})
+            or llm_output.get("usage_metadata", {})
+            or llm_output.get("usage", {})
+            or {}  # in case one is explicitly set to None
```
Comment (Yun-Kim) [inline ddtrace/llmobs/_integrations/langchain.py:500]: Are these mutually exclusive? i.e. llm_output will only have one of these filled while the others are None or don't exist? If so, thoughts on chaining the get default value instead of the OR operation?

```python
llm_output.get("token_usage", llm_output.get("usage_metadata", llm_output.get("usage", {})))
```

File: releasenotes/notes/llmobs-langchain-token-metrics-and-coloring-21eed854130b235c.yaml [hunk 4]
```diff
@@ -0,0 +1,5 @@
+---
+features:
+  - |
+    LLM Observability: When not using a provider integration (OpenAI, Anthropic, or Bedrock) with the LangChain integration, token metrics will be appended to the LLM Observability ``llm`` span.
+    LLM Observability: When ``chat_model.with_structured_output(..., method="json_mode")`` is used, or ``response_format={"type": "json_object"}`` is passed into the chat model invocation, the LLM Observability span will be an ``llm`` span instead of a ``workflow`` span.
```
Comment (Yun-Kim) [inline releasenotes/notes/llmobs-langchain-token-metrics-and-coloring-21eed854130b235c.yaml:5]:
```suggestion
    LLM Observability: When langchain's ``chat_model.with_structured_output(..., method="json_mode")`` is used, or ``response_format={"type": "json_object"}`` is passed into a langchain chat model invocation, the LLM Observability span will be an ``llm`` span instead of a ``workflow`` span.                 
```

**Review by Yun-Kim — APPROVED** [review]
Looks good for the most part, just a couple suggestions. Thanks @sabrenner!

File: ddtrace/llmobs/_integrations/langchain.py (hunk 2 above)
Comment (sabrenner) [reply ddtrace/llmobs/_integrations/langchain.py:107]: good question, probably not? I had just adjusted the logic from what we had before. I'll remove and see if it breaks tests

File: ddtrace/llmobs/_integrations/langchain.py (hunk 3 above)
Comment (sabrenner) [reply ddtrace/llmobs/_integrations/langchain.py:500]: Ah good point, it should only ever have one, the name just depends on the partner library. Let me change this logic (can maybe do this in a couple other spots), and see that it still looks good.

File: ddtrace/llmobs/_integrations/langchain.py (hunk 2 above)
Comment (sabrenner) [reply ddtrace/llmobs/_integrations/langchain.py]: looks like it's fine to remove

File: ddtrace/llmobs/_integrations/langchain.py (hunk 3 above)
Comment (sabrenner) [reply ddtrace/llmobs/_integrations/langchain.py]: i changed the logic a bit, but still added an extra `dict` check to be safe, since we're technically getting these from partner libraries, seemingly without standardization, I just wanna be extra safe we don't crash. 

Also, for some reason, sometimes these fields are explicitly set to `None`, so they don't fall back on the default value of `dict.get(..., default)`, so I added those extra `is not None` checks to be safe. 

tl;dr check everything to be safe with langchain partner libraries 😅
